### ISSUE TITLE
Items dissapearing from NextUp after a few days

### ISSUE THREAD

**Issue by fredrik-eriksson** [jellyfin/jellyfin#12367]
### This issue respects the following points:

- [X] This is a **bug**, not a question or a configuration issue; Please visit our forum or chat rooms first to troubleshoot with volunteers, before creating a report. The links can be found [here](https://jellyfin.org/contact/).
- [X] This issue is **not** already reported on [GitHub](https://github.com/jellyfin/jellyfin/issues?q=is%3Aopen+is%3Aissue) _(I've searched it)_.
- [X] I'm using an up to date version of Jellyfin Server stable, unstable or master; We generally do not support previous older versions. If possible, please update to the latest version before opening an issue.
- [X] I agree to follow Jellyfin's [Code of Conduct](https://jellyfin.org/docs/general/community-standards.html#code-of-conduct).
- [X] This report addresses only a single issue; If you encounter multiple issues, kindly create separate reports for each one.

### Description of the bug

I unfortunately lost track on exactly which version I started noticing this problem on, but for reasons I'm getting to it's probably v10.9.5. I started notice that new episodes did not show up in NextUp regardless of client (at least not in web client or Jellycon). But the behaviour was strange as the episodes would show up if I marked another episode in the series as unwatched and then watched again. And after a while I realized that stuff would only show up in NextUp if I had watched another episode in the series within a few days.

This happened regardless of what the setting "Max days in 'Next Up" in the webui was set to (it was set to 10 days).

Long story short: I experimented a bit with API calls using jellyfin-apiclient-python and found out that if I didn't send any "Limit" option in the NextUp API call I got all episodes I expected (5 of them at the time), but with "Limit" set to 20 I only received one episode for a very recently watched series:

```
# This gives 1 result
res = client.jellyfin.shows(
        '/NextUp',
        {
        "UserId": "{UserId}",
        "Limit": 20
        })

# This gives 5 results
res = client.jellyfin.shows(
        '/NextUp',
        {
        "UserId": "{UserId}",
        })
```

This is the reason I suspect PR #11956 and version 10.9.5 as it's the only recent change I can find when it comes to NextUp and the Limit property. My hypothesis is that the limit filtering is done too early in the SQL query(ies?). Perhaps that it filters on recently watched series instead of recently watched series with unwatched episodes?  That said; I extracted the sql queries from the logs (with and without a limit set) and to my untrained eye (and SQL-sense) I can't see what the problem is...

### Reproduction steps

Not 100% sure how to reproduce, but a guess would be:

1. Make sure you have a episode in NextUp
2. Add new episodes for *other series* and mark these as watched
3. The episode you had in NextUp will disappear eventually

### What is the current _bug_ behavior?

Not all expected episodes show up in NextUp (webui or API) when "Limit" filter is set.

### What is the expected _correct_ behavior?

"Limit" should be the maximum limit of NextUp episodes, but should have no effect unless the number of total NextUp episodes exceeds the limit.

### Jellyfin Server version

10.9.8+

### Specify the build version

10.9.8

### Environment

```markdown
- OS: Gentoo
- Linux Kernel: 6.6.35-gentoo-dist
- Virtualization: none
- Clients: Browser, API, Jellycon
- Browser: Firefox 115.13.0
- FFmpeg Version: 6.0.1-Jellyfin
- Playback Method:
- Hardware Acceleration:
- GPU Model:
- Plugins: Ani-Sync, AniDB, AniList, AudioDB, Fanart, Kodi Sync Queue, MusicBrainz, OMDb, Open Subtitles, Studio Images, TMDb, TVHeadend, TheTVDB, Trakt
- Reverse Proxy: nginx
- Base URL: none
- Networking: Host
- Storage: local
```

### Jellyfin logs

```shell
SQL log using limit 20:

aug 01 09:36:20 localhost jellyfin[1218105]: [09:36:20] [DBG] [8] Emby.Server.Implementations.Data.SqliteItemRepository: GetItemList query time (slow): 216.2577ms. Query: select type,data,EndDate,ChannelId,CommunityRating,IndexNumber,Width,Height,Name,Path,PremiereDate,ParentIndexNumber,ProductionYear,OfficialRating,RunTimeTicks,Size,DateModified,guid,ParentId,Audio,IsInMixedFolder,PrimaryVersionId,Album,LUFS,NormalizationGain,CriticRating,IsVirtualItem,SeriesName,SeasonName,SeasonId,SeriesId,ProviderIds,TotalBitrate,ExtraType,ExternalId,SeriesPresentationUniqueKey,OwnerId,UserDatas.UserId,UserDatas.lastPlayedDate,UserDatas.playbackPositionTicks,UserDatas.playcount,UserDatas.isFavorite,UserDatas.played,UserDatas.rating from TypedBaseItems A left join UserDatas on UserDataKey=UserDatas.Key And (UserId=@UserId) where type=@type AND TopParentId in ('b918fe2ac56551caeda1e78793660731','ed5bd7635a8509fd2a7c7bb22f7cfcce','27d78572387e7a0aec84e980aed31cad','e2b1dea58e98169b2560f5e3875e20fc','e5456d146846fe9b2b517bd9931bd504','232b36c96c8c161ffab816c9d81cc475','9c903d24948659f5d87b6638b01cd262','1071671e7bffa0532e930debee501d2e') Group by SeriesPresentationUniqueKey ORDER BY MAX(LastPlayedDate) DESC LIMIT 30


SQL log using no limit:

aug 01 09:34:17 localhost jellyfin[1218105]: [09:34:17] [DBG] [8] Emby.Server.Implementations.Data.SqliteItemRepository: GetItemList query time (slow): 253.284ms. Query: select type,data,EndDate,ChannelId,CommunityRating,IndexNumber,Width,Height,Name,Path,PremiereDate,ParentIndexNumber,ProductionYear,OfficialRating,RunTimeTicks,Size,DateModified,guid,ParentId,Audio,IsInMixedFolder,PrimaryVersionId,Album,LUFS,NormalizationGain,CriticRating,IsVirtualItem,SeriesName,SeasonName,SeasonId,SeriesId,ProviderIds,TotalBitrate,ExtraType,ExternalId,SeriesPresentationUniqueKey,OwnerId,UserDatas.UserId,UserDatas.lastPlayedDate,UserDatas.playbackPositionTicks,UserDatas.playcount,UserDatas.isFavorite,UserDatas.played,UserDatas.rating from TypedBaseItems A left join UserDatas on UserDataKey=UserDatas.Key And (UserId=@UserId) where type=@type AND TopParentId in ('b918fe2ac56551caeda1e78793660731','ed5bd7635a8509fd2a7c7bb22f7cfcce','27d78572387e7a0aec84e980aed31cad','e2b1dea58e98169b2560f5e3875e20fc','e5456d146846fe9b2b517bd9931bd504','232b36c96c8c161ffab816c9d81cc475','9c903d24948659f5d87b6638b01cd262','1071671e7bffa0532e930debee501d2e') Group by SeriesPresentationUniqueKey ORDER BY MAX(LastPlayedDate) DESC
```

**Comment by fredrik-eriksson** [comment]
Dug some more and it does indeed look like that query only returns the latest watched episodes, while figuring out which episodes should be "up next" is done [outside of SQL](https://github.com/jellyfin/jellyfin/blob/d4eeafe53ff8ae2f287f5dca49a873cd71f4c3da/Emby.Server.Implementations/TV/TVSeriesManager.cs#L187C79-L187C88).

So right now NextUp  (in webui) will not work if you watch more than 30-ish episodes in between one episode in a series to the next (seems the webui queries with "limit=24", which for some reason results in an SQL query using [LIMIT 34](https://github.com/cptn-x/jellyfin/blob/52aded8e41800ae21392609d8b88bca0c02b7172/Emby.Server.Implementations/TV/TVSeriesManager.cs#L111)), and using limit in the API will yield unexpected results. Reverting #11956 is probably the easiest fix (with the obvious drawback that the slowness it attempted to fix will return). Someone smarter than me will have to figure out any other solution :slightly_smiling_face:

**Comment by fredrik-eriksson** [comment]
Fixed in 10.9.10
